datasets/at/meine_abgeordneten/crawler.py

```diff
@@ -14,22 +14,21 @@
 MAX_POSITION_NAME_LENGTH = 120
 
 
-def extract_dates(context, url, el):
+def extract_dates(context: Context, url, el):
     active_date_el = el.find('.//span[@class="aktiv"]')
     inactive_dates_el = el.find('.//span[@class="inaktiv"]')
     if active_date_el is not None:
-        start_date = h.parse_date(
-            active_date_el.text_content().replace("seit ", ""),
-            context.dataset.dates.formats,
+        start_date = h.extract_date(
+            context.dataset, active_date_el.text_content().replace("seit ", "")
         )
         end_date = None
         assume_current = True
     elif inactive_dates_el is not None:
-        inactive_dates = inactive_dates_el.text_content()
+        inactive_dates = inactive_dates_el.text_content().replace("ab ", "")
         if " - " in inactive_dates:
             start_date, end_date = inactive_dates.split(" - ")
-            start_date = h.parse_date(start_date, context.dataset.dates.formats)
-            end_date = h.parse_date(end_date, context.dataset.dates.formats)
+            start_date = h.extract_date(context.dataset, start_date.strip())
+            end_date = h.extract_date(context.dataset, end_date.strip())
```

Comment:
if we leave the parsing to make_occupancy() and move the `? -> None` to lookups, can we get the same output? that'd be a nice big cleanup.

I think the real logic in this function is about setting assume_current true/false. But it doesn't look like parsing the dates according to our date formats is actually part of that logic, unless I missed something